tests/api/v1/collectorGroups/put.js

```diff
@@ -203,4 +236,254 @@ describe('tests/api/v1/collectorGroups/put.js >', () => {
+      generatorInst.reload()
+      .then((updatedGenInst) => {
+        expect(updatedGenInst.currentCollector).to.not.exist;
+        api.put(`/v1/collectorGroups/${cg.name}`)
+        .set('Authorization', token)
+        .send({
+          name: cg.name,
+          collectors: [collector1.name, collector2.name],
+        })
+        .expect(httpStatus.OK)
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
+
+          const collectors = res.body.collectors;
+          expect(Array.isArray(collectors)).to.be.true;
+          expect(collectors.length).to.equal(2);
+          const collectorNames = collectors.map((collector) => collector.name);
+          expect(collectorNames).to.have.members([collector1.name, collector2.name]);
+
+          api.get(`/v1/generators/${generatorInst.name}`)
+          .set('Authorization', token)
+          .expect(httpStatus.OK)
+          .end((err, res) => {
+            if (err) {
+              return done(err);
+            }
+
+            expect(res.body.currentCollector).to.not.exist;
+            return done();
+          });
+        });
+      });
+    });
+
+    it('already assigned generator, put collectorGroup should not change ' +
+      'currentCollector if currentCollector exists in updated collector list',
+      (done) => {
+        generatorInst.update({ collectorId: collectorAlive1.id })
+        .then((gen) => gen.reload())
+        .then((updatedGenInst) => {
+          expect(updatedGenInst.currentCollector.name).to.be.equal(collectorAlive1.name);
+          api.put(`/v1/collectorGroups/${cg.name}`)
+          .set('Authorization', token)
+          .send({
+            name: cg.name,
+            collectors: [collector1.name, collectorAlive1.name, collectorAlive2.name],
+          })
+          .expect(httpStatus.OK)
+          .end((err, res) => {
+            if (err) {
+              return done(err);
+            }
+
+            const collectors = res.body.collectors;
+            expect(Array.isArray(collectors)).to.be.true;
+            expect(collectors.length).to.equal(3);
+            const collectorNames = collectors.map((collector) => collector.name);
+            expect(collectorNames).to.have.members(
+              [collector1.name, collectorAlive1.name, collectorAlive2.name]
+            );
+
+            api.get(`/v1/generators/${generatorInst.name}`)
+            .set('Authorization', token)
+            .expect(httpStatus.OK)
+            .end((err, res) => {
+              if (err) {
+                return done(err);
+              }
+
+              expect(res.body.currentCollector.name).to.equal(collectorAlive1.name);
+              return done();
+            });
+          });
+        }).catch(done);
+      });
+
+    it('already assigned generator, put collectorGroup should set ' +
+      'currentCollector to null if currentCollector does not exist in ' +
+      'updated collector list', (done) => {
```

Comment:
How about another similar test but not including collectors attribute in collectorGroups put request?
```
        api.put(`/v1/collectorGroups/${cg.name}`)
        .set('Authorization', token)
        .send({
          name: cg.name,
        })
```

Reply:
added.